vendor/engines/ldap_authentication/lib/ldap_authentication/user_converter.rb

```diff
@@ -0,0 +1,23 @@
+module LdapAuthentication
+
+  class UserConverter
+
+    CONVERTABLE_ATTRIBUTES = [:username, :first_name, :last_name, :email].freeze
+
+    def initialize(user_entry)
+      @user_entry = user_entry
+    end
+
+    def to_user
+      ::User.new(attributes)
+    end
+
+    def attributes
+      CONVERTABLE_ATTRIBUTES.each_with_object({}) do |field, output|
+        output[field] = @user_entry.public_send(field)
+      end
+    end
```

Comment:
Nevermind, I was confused a little. The result of `each_with_object` is the last expression, that's what's returned.